client/src/partials/section_bilan/section_bilan.js

```diff
@@ -1,77 +1,111 @@
+// TODO Handle HTTP exception errors (displayed contextually on form)
 angular.module('bhima.controllers')
-.controller('sectionBilan', [
-  '$scope',
-  '$translate',
-  'validate',
-  'connect',
-  'messenger',
-  function ($scope, $translate, validate, connect, messenger) {
-    var dependencies = {},
-        session = $scope.session = {};
-
-    dependencies.sectionBilans = {
-      query : {
-        identifier : 'id',
-        tables : {
-          'section_bilan' : {
-            columns : ['id', 'text', 'is_actif', 'position']
-          }
-        }
-      }
-    };
-
-    function startup (models) {
-      angular.extend($scope, models);
-    }
-
-    validate.process(dependencies)
-    .then(startup);
-
-    $scope.doTranslate = function (key){
-      return $translate.instant(key);
-    };
-
-    $scope.delete = function (sectionBilan) {
-      connect.delete('section_bilan', 'id', sectionBilan.id)
+.controller('sectionBilanController', sectionBilanController);
+
+sectionBilanController.$inject = [
+  'SectionBilanService', '$window', '$translate'
+];
+
+function sectionBilanController(sectionBilanService, $window, $translate) {
+  var vm = this;
+  vm.view = 'default';
+
+  // bind methods
+  vm.create = create;
+  vm.submit = submit;
+  vm.update = update;
+  vm.cancel = cancel;
+  vm.del    = del;  
+
+  vm.doTranslate = doTranslate;
+
+  function handler(error) {
+    console.error(error);
+  }
+
+  // fired on startup
+  function startup() {
+    // start up loading indicator
+    vm.loading = true;
+
+    // load sections bilans
+    refreshSectionBilans();
+  }
+
+  function cancel() {
+    vm.view = 'default';
+  }
+  
+
+  function create() {
+    vm.view = 'create';
+    vm.sectionBilan = {};    
+  }
+
+  // switch to update mode
+  // data is an object that contains all the information of a Section Bilan
+  function update(data) {
+    vm.view = 'update';
+    vm.sectionBilan = data;
+  }
+
+  function doTranslate(key){
+    return $translate.instant(key);
+  }
+  
+  // refresh the displayed Sections Bilans
+  function refreshSectionBilans() {
+    return sectionBilanService.read(null,{ detailed : 1 }).then(function (data) {
+      vm.sectionBilans = data;
+      vm.loading = false;
+    });
+  }
+
+  // form submission
+  function submit(invalid) {
+    if (invalid) { return; }
+
+    var promise;
+    var creation = (vm.view === 'create');
+
+    var sectionBilan = angular.copy(vm.sectionBilan);
+    
+    promise = (creation) ?
+      sectionBilanService.create(sectionBilan) :
```

Comment:
State based implementations are currently being re-factored. This will not block the pull request however it will need to refined once the routing method is set. 

This is discussed [here](https://github.com/IMA-WorldHealth/bhima-2.X/pull/200#issuecomment-200070748) and tracked [here](https://github.com/IMA-WorldHealth/bhima-2.X/pull/200#issuecomment-200070748)